src/components/Contentful/HomePage/NewHomeLoansCardCarousel.vue

```diff
@@ -61,82 +68,12 @@ export default {
 		/**
 		 * Extract Loan Channel settings from Contentful Ui Setting dataObject
 		* */
-		contentfulLoanChannels() {
-			const uiSetting = this.content?.contents?.find(({ contentType }) => {
-				return contentType ? contentType === 'uiSetting' : false;
-			});
-			return uiSetting?.dataObject?.loanChannels ?? [];
-		},
-		/**
-		 * Extract Loan Display settings from Contentful Ui Setting dataObject
-		* */
-		loanDisplaySettings() {
-			const uiSetting = this.content?.contents?.find(({ contentType }) => {
-				return contentType ? contentType === 'uiSetting' : false;
-			});
-			return {
-				loanLimit: uiSetting?.dataObject?.loanLimit ?? 1,
-				showViewMoreCard: uiSetting?.dataObject?.showViewMoreCard ?? false
-			};
-		},
-		combinedLoanChannelData() {
-			return this.contentfulLoanChannels.map(channel => {
-				const matchedLoanChannel = this.loanChannelData.find(lc => lc.id === channel.id);
-				return { ...matchedLoanChannel, ...channel };
-			});
-		},
-		loanChannelIds() {
-			return this.contentfulLoanChannels.map(channelSetting => {
-				return channelSetting.id;
-			});
-		},
-		loanQueryLimit() {
-			return this.loanDisplaySettings?.loanLimit ?? 1;
-		},
-		selectedChannelLoanIds() {
-			const selectedChannel = this.combinedLoanChannelData.find(channel => {
-				return this.selectedChannel?.id === channel.id;
-			});
-			return selectedChannel?.loans?.values?.map(loan => loan.id) ?? [];
-		},
-		showViewMoreCard() {
-			return this.loanDisplaySettings?.showViewMoreCard ?? false;
-		}
 	},
-	mounted() {
-		this.fetchLoanChannel();
-	},
-	methods: {
-		fetchLoanChannel() {
-			this.apollo.query({
-				query: gql`query selectedLoanCategory($loanChannelIds: [Int]!, $loanLimit: Int) {
-					lend {
-						loanChannelsById(ids: $loanChannelIds){
-							id
-							name
-							url
-							loans(limit: $loanLimit) {
-								values {
-									id
-								}
-							}
-						}
-					}
-				}`,
-				variables: {
-					loanChannelIds: this.loanChannelIds,
-					loanLimit: this.loanQueryLimit
-				},
-			}).then(result => {
-				// Set All Active Loan Channels Data
-				const loanChannels = result?.data?.lend?.loanChannelsById ?? [];
-				this.loanChannelData = loanChannels;
-				// Activate the first channel available
-				const initialChannel = this.combinedLoanChannelData[0];
-				this.selectedChannel = initialChannel;
-				// Make the carousel visible
-				this.showCarousel = true;
-			});
+	watch: {
+		loans() {
+			if (this.loans.length > 0 && this.selectedChannel.id) {
```

Comment:
Why are the loans being passed in, but only the length checked? Can we just pass in the `selectedChannel` to this component?